### DISCUSSION TITLE
Semantic approach to HTML & CSS

### DISCUSSION THREAD

**Discussion by JensRoland** [nuejs/nue#32] [Q&A] [answered]
One thing I would love to see from this project is a return to the semantic HTML and CSS that we had before utility classes and Tailwind reinvented the `<font>` tag and the `style="color:red;"` attribute.

Back in the day, we had the CSS Zen Garden. These days, we get truly inspired designs on https://stylestage.dev/

I would love to see a component library based on completely semantic/dogmatic HTML and a CSS library/toolchain that made theming of semantic HTML more intuitive and less fiddly while providing a minimal and optional/modular set of utility classes to simplify the boilerplate of grid layouts, responsive elements, typography, and animations. Something minimal in the vein of https://semantic-ui.com/ meets https://get.foundation/ is what I'm hoping for.

**Comment by tipiirai** [comment, marked as answer]
Looove to hear this! One of the main missions of Nue is to bring back the semantic approach to HTML and CSS. My goal is to first build something like Headless UI, but with Nue and semantic HTML. And together with Nue MVC, I'm planning to release a demo admin dashboard using the same philosophy. Finally, after Nuekit is complete it's time for a modern take on Zen Garden. This is probably going to get 6+ months, but I super committed to this.

**Reply by JensRoland** [reply]
That's awesome! Headless is probably the way to go, good call.

As for admin dashboards, personally I like CoreUI (https://coreui.io/product/free-bootstrap-admin-template/) and would love to see a CoreUI implemented with Nue JS / Nue UI.

**Reply by tipiirai** [reply]
Note that there is no CSS framework planned on the roadmap: https://nuejs.org/tools/ 

The goal of Nue is to provide examples and best practices on how to work closer to standards and to bring the [UX developer](https://css-tricks.com/the-great-divide/) back to the forefront. There will be an admin dashboard and lot of other examples, that are minimalistic, look good, and use the semantic approach. But they are just examples, without using any official Nue CSS framework. I'm hoping that these would eventually come from the community.
